src/compose/service.ts

```diff
@@ -1124,26 +1147,60 @@ export class Service {
 	): ServiceConfig['volumes'] {
 		let volumes: ServiceConfig['volumes'] = [];
 
-		_.each(composeVolumes, (volume) => {
-			const isBind = _.includes(volume, ':');
-			if (isBind) {
-				const [bindSource, bindDest, mode] = volume.split(':');
-				if (!path.isAbsolute(bindSource)) {
-					// namespace our volumes by appId
-					let volumeDef = `${appId}_${bindSource.trim()}:${bindDest.trim()}`;
+		// namespace our volumes by appId
+		const namespaceVolume = (volumeSource: string) =>
+			`${appId}_${volumeSource.trim()}`;
+
+		for (const volume of composeVolumes || []) {
+			let isBindMount = false;
+
+			if (
+				(LongSyntaxVolume.is(volume) && volume.type === 'tmpfs') ||
+				(LongSyntaxVolume.is(volume) &&
+					volume.type === 'volume' &&
+					!volume.source) ||
+				(typeof volume === 'string' && !volume.includes(':'))
+			) {
+				// Handle 'tmpfs' long syntax volumes, 'volume' long syntax volumes without a name, and anonymous volumes.
+				// Anonymous volumes can only be set by using the `VOLUME` instruction inside a Dockerfile.
+				// Tmpfs volumes do not require namespacing by appId.
+				volumes.push(volume);
+			} else if (
+				LongSyntaxVolume.is(volume) &&
+				(volume.type === 'bind' || volume.type === 'volume')
+			) {
+				// Handle long syntax volumes of type 'volume' or 'bind'.
+				// Other invalid mount types will raise a compose error,
+				// so we don't need to worry about other long syntax types.
+				// Volumes are added to config while binds are ignored and flagged for console warning.
+				isBindMount = path.isAbsolute(volume.source!);
```

Comment:
Clarified here @pipex -- this look okay?

Reply:
On second thought, this all looks to be a bit too verbose.. how do we feel about making the `io-ts` long syntax types more specific, i.e. `LongSyntaxTmpfs`, `LongSyntaxVolume`, `LongSyntaxBind`? I'm asking this because, as it turns out, the options and required configs for these different long syntax types turn out to be more different than the same. Their only similarities are that they require `target` and `type` fields. We could even add `ShortSyntaxVolume`. It would make the code here more terse which is a good thing.

Reply:
I like that idea, this definitely is looking too verbose. Using the io-ts type guards will also help you avoid the non-null assertions `volume.source!`

Reply:
Latest push switches completely to using io-ts `brand` type refinements.